File: typescript/202-happy-number.ts
function getSquaredSum(n: number): number {
    let squaredSum = 0
    while (n > 0) {
        squaredSum += (n%10)**2
        n = Math.floor(n/10)
    }
    return squaredSum
}

function isHappy(n: number): boolean {
    let slow = n
    let fast = getSquaredSum(n)
    while (slow !== fast) {
        if (slow === 1) return true
        slow = getSquaredSum(slow)
        fast = getSquaredSum(getSquaredSum(fast))
    }
    return slow === 1
};

File: golang/202-happy-number.go
package golang

func getSquaredSum(n int) int {
    squaredSum := 0
    for n > 0 {
        squaredSum += (n%10) * (n%10)
        n = n/10
    }
    return squaredSum
}

func isHappy(n int) bool {
    slow := n
    fast := getSquaredSum(n)
    for slow != fast {
        if fast == 1 {
            return true
        }
        slow = getSquaredSum(slow)
        fast = getSquaredSum(getSquaredSum(fast))
    }
    return fast == 1
}